src/index.js

```diff
@@ -19,30 +24,52 @@ class Dropzone extends React.Component {
   constructor(props, context) {
     super(props, context);
     this.onClick = this.onClick.bind(this);
+    this.onDocumentDrop = this.onDocumentDrop.bind(this);
     this.onDragStart = this.onDragStart.bind(this);
     this.onDragEnter = this.onDragEnter.bind(this);
     this.onDragLeave = this.onDragLeave.bind(this);
     this.onDragOver = this.onDragOver.bind(this);
     this.onDrop = this.onDrop.bind(this);
     this.onFileDialogCancel = this.onFileDialogCancel.bind(this);
     this.fileAccepted = this.fileAccepted.bind(this);
+    this.setRef = this.setRef.bind(this);
     this.isFileDialogActive = false;
     this.state = {
       isDragActive: false
     };
   }
 
   componentDidMount() {
-    this.enterCounter = 0;
+    const { allowGlobalDrops } = this.props;
+    this.dragTargets = [];
+
+    if (!allowGlobalDrops) {
+      document.addEventListener('dragover', Dropzone.onDocumentDragOver, false);
```

Comment:
If `dragover` is not provided, dropped items will still take over the browser window, because the browser doesn't consider a div with only a `drop` event (and no `dragover`) to be a valid `drop` target.